### ISSUE TITLE
Calculating gradient wrt input

### ISSUE THREAD

**Issue by ayush1999** [keras-team/keras#11909]
I have a pretrained keras model. I wish to calculate the gradient of the loss function of the model wrt an input tensor. (I'm trying to implement FGSM attack). However, using `K.gradients` gives me a None value. Why is it so? Also, can someone point me to existing FGSM attack code for keras?

**Comment by msymp** [comment]
@ayush1999 , Please study the following Stack Overflow post where Matias Valdenegro's answer clearly states how to use the _K.gradients_  ;
Also here is the Keras FGSM example code working over MNIST:
https://github.com/tensorflow/cleverhans/blob/master/cleverhans_tutorials/mnist_tutorial_keras_tf.py
This issue is closed now.
Thanks.

**Comment by sajed-zarrinpour** [comment]
I have a multi-input/multi-output model and I believe this will help you:
`g = K.gradients(model.get_output_at(0)[index_of_the_desired_out_put], model.get_input_at(0)[index_of_the_desired_desired_input])
`
though what I am after is to get the second gradient of the model w.r.t. it's inputs. the problem is that the evaluation of the gradient is zero vector. the code for that is
`gg = K.gradients(g, model.get_input_at(0)[0])
a = sess.run(gg,feed_dict={model.get_input_at(0)[0]: x_train,model.get_input_at(0)[1]:a_train})`

anyone can help me?
I am searching. If anyone knows how to do it properly I would be glad.
thank you all.

**Comment by sajed-zarrinpour** [comment]
I've found the trick! I was using **ReLu** as my activation function which causes the whole network to be linear (a first-order polynomial indeed). changing the activation function to something like **tanh** would work.
